### PR TITLE
Feature/#571 more button for replays

### PR DESCRIPTION
**Pull request by axel1200**

### CODE REVIEW

File: src/main/java/com/faforever/client/map/MapVaultController.java [hunk 1]
```diff
@@ -283,6 +282,7 @@ private void appendSearchResult(List<MapBean> maps, Pane pane) {
   }
 
   private void displayMapsFromSupplier(Supplier<CompletableFuture<List<MapBean>>> mapsSupplier) {
+    currentPage = 0;
```
Comment (axel1200) [inline src/main/java/com/faforever/client/map/MapVaultController.java]: setting this here means that you can also click the more button if it was getNewest Replays and not only with a query

File: src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java [hunk 2]
```diff
@@ -74,39 +80,46 @@ public void initialize() {
...
+      }
 
       replays.forEach(replay -> {
         ReplayCardController controller = uiService.loadFxml("theme/vault/replay/replay_card.fxml");
         controller.setReplay(replay);
         controller.setOnOpenDetailListener(this::onShowReplayDetail);
         children.add(controller.getRoot());
 
-        if (replays.size() == 1) {
+        if (replays.size() == 1&&!append) {
```
Comment (micheljung) [inline src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java]: Please enable code formatting in the commit dialog :-)

File: src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java [hunk 3]
```diff
@@ -182,16 +194,35 @@ private void refresh() {
...
+    mapsSupplier.get()
+        .thenAccept(this::displaySearchResult)
+        .exceptionally(e -> {
+          notificationService.addNotification(new ImmediateNotification(i18n.get("errorTitle"),
+              i18n.get("vault.replays.searchError"), Severity.ERROR, e,
+              Collections.singletonList(new DismissAction(i18n))));
+          enterShowroomState();
+          return null;
+        });
   }
```
Comment (micheljung) [inline src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java]: This file needs to be formatted correctly, too. Probably every file in the PR :D

Comment (axel1200) [reply src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java]: ok

**Comment by axel1200** [comment]
The reformatter was on maybe the setting are wrong

File: src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java [hunk 4]
```diff
@@ -138,25 +150,24 @@ private void enterSearchingState() {
...
     enterSearchingState();
-    replayService.findByQuery(query, MAX_SEARCH_RESULTS)
-        .thenAccept(this::displaySearchResult)
-        .exceptionally(e -> {
-          notificationService.addNotification(new ImmediateNotification(i18n.get("errorTitle"),
-              i18n.get("vault.replays.searchError"), Severity.ERROR, e,
-              Collections.singletonList(new DismissAction(i18n))));
-          return null;
-        });
+    displayReplaysFromSupplier(() ->replayService.findByQuery(query, MAX_SEARCH_RESULTS,currentPage++) );
```
Comment (micheljung) [inline src/main/java/com/faforever/client/replay/OnlineReplayVaultController.java]: format file

File: src/main/java/com/faforever/client/api/MockFafApiAccessor.java [hunk 5]
```diff
@@ -160,22 +160,22 @@ public Mod getMod(String uid) {
   }
 
   @Override
-  public List<Game> getNewestReplays(int count) {
+  public List<Game> getNewestReplays(int count,int page) {
```
Comment (micheljung) [inline src/main/java/com/faforever/client/api/MockFafApiAccessor.java]: Please enable the auto-formatting in your commit settings, or if you have done it already, reformat this file :)

File: src/main/java/com/faforever/client/replay/ReplayService.java [hunk 6]
```diff
@@ -22,13 +22,13 @@
...
+  CompletableFuture<List<Replay>> getNewestReplays(int topElementCount,int page);
 
-  CompletableFuture<List<Replay>> getHighestRatedReplays(int topElementCount);
+  CompletableFuture<List<Replay>> getHighestRatedReplays(int topElementCount,int page);
 
-  CompletableFuture<List<Replay>> getMostWatchedReplays(int topElementCount);
+  CompletableFuture<List<Replay>> getMostWatchedReplays(int topElementCount,int page);
 
-  CompletableFuture<List<Replay>> findByQuery(String condition, int maxResults);
+  CompletableFuture<List<Replay>> findByQuery(String condition, int maxResults,int page);
```
Comment (micheljung) [inline src/main/java/com/faforever/client/replay/ReplayService.java]: format file

File: src/main/java/com/faforever/client/replay/ReplayServiceImpl.java [hunk 7]
```diff
@@ -274,23 +274,23 @@ public void runReplay(Integer replayId) {
...
-  public CompletableFuture<List<Replay>> getMostWatchedReplays(int topElementCount) {
-    return fafService.getMostWatchedReplays(topElementCount);
+  public CompletableFuture<List<Replay>> getMostWatchedReplays(int topElementCount,int page) {
+    return fafService.getMostWatchedReplays(topElementCount,page);
   }
 
   @Override
-  public CompletableFuture<List<Replay>> findByQuery(String query, int maxResults) {
-    return fafService.findReplaysByQuery(query, maxResults);
+  public CompletableFuture<List<Replay>> findByQuery(String query, int maxResults,int page) {
```
Comment (micheljung) [inline src/main/java/com/faforever/client/replay/ReplayServiceImpl.java]: format file

**Comment by axel1200** [comment]
I tried again, if it does not work I might need your help getting my reformatter correctly set

**Pull request merged**